src/context/ThemeContext.jsx

```diff
@@ -0,0 +1,31 @@
+import { createContext, useContext, useState, useEffect } from "react";
+
+const ThemeContext = createContext();
+
+const ThemeProvider = ({ children }) => {
+  const preferDarkMode = matchMedia("(prefers-color-scheme: dark)").matches;
+  const localMode = JSON.parse(localStorage.getItem("mode"));
+  const [darkMode, setDarkMode] = useState(localMode ?? preferDarkMode);
+  useEffect(() => {
+    localStorage.setItem("mode", darkMode);
+
+    return () => {
+      localStorage.clear("mode");
```

Comment:
Is it required to write the statement & return the function here?

Reply:
I think this is the way to write cleanup functions